### PR TITLE
Update GEPA Thought Block Validation to Include Signature Field

### PR DESCRIPTION
**Pull request by anndvision**
Updates the GEPA example validation logic to check all three fields (`text`, `signature`, `summary`) of `Thought` blocks instead of just `text` and `summary`. Examples are now only dropped if ALL THREE fields are `None`.

This allows GEPA to work with encrypted thinking responses that only contain a `signature` field.

---

> [!NOTE]
> Adjusts GEPA example validation to consider `Thought.signature`, enabling encrypted-thinking responses where only `signature` is present.
> 
> - Update validation in `validate.rs` for both `stored_output` and `stored_input.messages` to reject `Thought` only when `text`, `signature`, and `summary` are all `None`
> - Destructure `Thought` fields to trigger compile errors if the struct changes
> - Refresh doc comments and error messages accordingly
> - Add/rename unit tests: cases for all-none invalid and signature-only valid for both stored output and input
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit de43359cb3d499daa06d0d5ca210bc8ba33e63c6. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

File: tensorzero-optimizers/src/gepa/validate.rs [hunk 1]
```diff
@@ -193,9 +193,12 @@ fn validate_stored_output(stored_output: Option<&StoredOutput>) -> Result<(), St
                 }
                 // Check Thought block
                 ContentBlockChatOutput::Thought(thought) => {
-                    if thought.text.is_none() && thought.summary.is_none() {
```
[the remark this answers is not in the document]
Comment (anndvision) [reply tensorzero-optimizers/src/gepa/validate.rs]: thx
